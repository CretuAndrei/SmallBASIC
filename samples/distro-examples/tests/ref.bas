' reference variable tests

a = "cat"
b = @a

print "cat=", b
a = "dog"
print "dog=", b
print "3=", len(b)
print "1=", not empty(b)
print "a=b", iFF(a==b, "a=b", "a<>b")

b = "goodbye a"
print "a<>b", iFF(a==b, "a=b", "a<>b")

dim rooms
sub addRoom(byref room)
  rooms << @room
end

dim kitchen,hall,toilet
kitchen.name= "kitchen"
hall.name = "hall"
toilet.name ="toilet"

addRoom(kitchen)
addRoom(hall)

print rooms(0)
print rooms(1)

kitchen.name = "Kitchen"
kitchen.fridge = "empty"
print rooms(0)

insert rooms, 0, @toilet
toilet.occupied = true

print rooms(0)
print rooms(1)
print rooms(2)

