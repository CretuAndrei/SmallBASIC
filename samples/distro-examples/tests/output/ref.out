cat=	cat
dog=	dog
3=	3
1=	1
a=b	a=b
a<>b	a<>b
{"NAME":"kitchen"}
{"NAME":"hall"}
{"FRIDGE":"empty","NAME":"Kitchen"}
{"NAME":"toilet","OCCUPIED":-1}
{"FRIDGE":"empty","NAME":"Kitchen"}
{"NAME":"hall"}
toilet
